### PR TITLE
drivers: sensor: tmp1075: Add tmp1075 sensor driver and sample

### PR DESCRIPTION
**Pull request by Paplewski**
TI tmp1075 driver implemented based on tmp108 driver. The driver initializes the sensor based on the DTS. See sample app.overlay.

Added sample to show example driver usage.
To see default DTS configuration option inspect `ti,tmp1075.yaml` bindings file.

### CODE REVIEW

File: drivers/sensor/ti/tmp1075/tmp1075.c [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+#ifdef CONFIG_TMP1075_ALERT_INTERRUPTS
+static int setup_interrupts(const struct device *dev)
+{
+	struct tmp1075_data *drv_data = dev->data;
+	const struct tmp1075_config *config = dev->config;
+	const struct gpio_dt_spec *alert_gpio = &config->alert_gpio;
+	int result;
+
+	if (!device_is_ready(alert_gpio->port)) {
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp1075/tmp1075.c:172]: Use `gpio_is_ready_dt`

File: drivers/sensor/ti/tmp1075/tmp1075.c [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+#endif
+
+int tmp1075_init(const struct device *dev)
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp1075/tmp1075.c:202]:
```suggestion
static int tmp1075_init(const struct device *dev)
```

File: drivers/sensor/ti/tmp1075/tmp1075.c [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+{
+	const struct tmp1075_config *cfg = dev->config;
+	struct tmp1075_data *data = dev->data;
+
+	if (!device_is_ready(cfg->bus.bus)) {
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp1075/tmp1075.c:207]: Use `i2c_is_ready_dt`

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+{
+	if (!device_is_ready(dev)) {
+		return -ENODEV;
+	}
+
+	struct tmp1075_data *drv_data = dev->data;
+
+	if (trig->type == SENSOR_TRIG_THRESHOLD) {
+		drv_data->temp_alert_handler = handler;
+		drv_data->temp_alert_trigger = *trig;
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmp1075/tmp1075_trigger.c:49]: Save the address of the trigger, not a copy of the trigger.

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+compatible: "ti,tmp1075"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  conversion-rate:
+    description: Conversion rate in ms.
+    type: int
+    default: 220
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,tmp1075.yaml:16]: Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    enum:
+      - 275 # 27.5ms
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,tmp1075.yaml:18]: This doesn't make sense

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    description: |
+      Identifies the ALERT signal, which is active-low open drain when
+      produced by the sensor.
+  one-shot-conversion:
+    description: One-shot conversion mode.
+    type: boolean
+  consecutive-fault-measurements:
+    description: Number of consecutive measured faults that will trigger the alert.
+    type: int
+    default: 4
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,tmp1075.yaml:33]: Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  interrupt-mode:
+    description: Selects the function of the ALERT pin.
+    type: boolean
+  shutdown-mode:
+    description: Sets the device in shutdown mode to conserve power.
+    type: boolean
+  lower-threshold:
+    description: Lower threshold for alert interrupt. Expressed in degrees C.
+    type: int
+    default: 28
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,tmp1075.yaml:51]: Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  upper-threshold:
+    description: Upper threshold for alert interrupt. Expressed in degrees C.
+    type: int
+    default: 29
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ti,tmp1075.yaml:55]: Need to explain in the description why the default value was selected
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please use the existing `samples/sensor/thermometer` instead of introducing a driver-specific sample.

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 5 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: removed default value and updated overlay file

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 6 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: removed default value and updated overlay file

Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: changed to microseconds

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 7 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: removed default value and updated overlay file

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 8 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: removed default value and updated overlay file

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 9 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: removed default value and updated overlay file

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c (hunk 4 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075_trigger.c]: done

File: drivers/sensor/ti/tmp1075/tmp1075.c (hunk 3 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075.c]: done

File: drivers/sensor/ti/tmp1075/tmp1075.c (hunk 2 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075.c]: updated

File: drivers/sensor/ti/tmp1075/tmp1075.c (hunk 1 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075.c]: done

File: tests/drivers/build_all/sensor/i2c.dtsi [hunk 10]
```diff
@@ -1113,3 +1113,14 @@ test_i2c_mmc56x3: mmc56x3@92 {
 	magn-odr = <0>;
 	auto-self-reset;
 };
+
+test_i2c_tmp1075: tmp1075@91 {
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/i2c.dtsi:1117]: The address should be 0x98. The previous node is incorrect and should be 0x97, but that can be fixed in a separate patch.

Comment (Paplewski) [reply tests/drivers/build_all/sensor/i2c.dtsi]: updated both addresses in a separate commit

**Review by MaureenHelm — APPROVED** [review]

File: drivers/sensor/ti/tmp1075/Kconfig [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+	  Enable the driver for Texas Instruments TMP1075 High-Accuracy Digital
+	  Temperature Sensors.
+
+if TMP1075
+
+config TMP1075_ALERT_INTERRUPTS
+	bool "Allow interrupts to service over and under temp alerts"
+	help
+	  This will set up interrupts to service under and over temp alerts
+	  see TMP1075 spec sheet for more information on how these work.
```
Comment (ubieda) [inline drivers/sensor/ti/tmp1075/Kconfig:22]: May we apply this pattern enforced in this (on-going) PR? #79226

**Review by ubieda — APPROVED** [review]

**Pull request merged**